### PR TITLE
Ecm 610

### PR DESCRIPTION
**Pull request by gauravhmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CFTS-1773
https://tools.hmcts.net/jira/browse/ECM-610

### Change description ###

Update judges' code

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@RequiredArgsConstructor
+@Slf4j
+public class RefDataFixesCcdDataSource implements RefDataFixesDataSource {
+
+    private final String authToken;
+
+    @Override
+    public List<SubmitEvent> getData(String caseTypeId, String dateFrom, String dateTo, CcdClient ccdClient) {
+        try {
+            var query = RefDataFixesElasticSearchQuery.create(
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java:18]: don't know what we are doing with var in the ethos project :)

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                    dateFrom, dateTo);
+            return ccdClient.executeElasticSearch(authToken, caseTypeId, query);
+        } catch (Exception e) {
+            throw new RefDataFixesException(String.format(
+                    "Failed to get hearings by hearing type search results for case type id %s", caseTypeId), e);
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java:23]: error message?

Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java]: changed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java (hunk 1 above)
Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/RefDataFixesCcdDataSource.java]: removed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//
+//        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+//            log.error("Invalid Token {}", userToken);
+//            return ResponseEntity.status(FORBIDDEN.value()).build();
+//        }
+//
+//        RefDataFixesData caseData = referenceDataFixesService.insertClaimServedDate(refDataFixesRequest.getCaseDetails(), userToken);
+//
+//        return getCallbackRespEntityNoErrors(caseData);
+//    }
```
Comment (btensay) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java:86]: Any reason why the commented section be removed?

Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java:86]: It is other half of the ticket, not yet tested, so commented out

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+        RefDataFixesCcdDataSource dataSource = new RefDataFixesCcdDataSource(userToken);
+        RefDataFixesData caseData = referenceDataFixesService.updateJudgesItcoReferences(
+                refDataFixesRequest.getCaseDetails(), userToken, dataSource);
+
+        return getCallbackRespEntityNoErrors(caseData);
+    }
+
+//    @PostMapping(value = "/insertClaimServedDate", consumes = APPLICATION_JSON_VALUE)
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java:63]: this commented out code has nothing to do with the change

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                    ccdClient.submitEventForCase(authToken, caseData, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), returnedRequest, String.valueOf(submitEvent.getCaseId()));
+                }
+                log.info(String.format(
+                        "Existing Judge's code in all cases from %s to %s is " +
+                                "updated to required judge's code", dateFrom, dateTo));
+            }
+            return refDataFixesData;
+        } catch (Exception ex) {
+            log.error(MESSAGE + refDataFixesDetails.getCaseId(), ex);
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java:59]: Logging and throwing is an anti-pattern. What do you see in the logs by doing this? Do you get the same thing twice?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDate(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        } else {
+            dateFrom = LocalDate.parse(refDataFixesData.getDateFrom(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDateTo(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        }
+        return new ArrayList<>(List.of(dateFrom, dateTo));
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java:80]: You don't need to create an ArrayList. What is wrong with returning the value from List.of ?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+            HearingType hearingType = hearingTypeItem.getValue();
+            if (!Strings.isNullOrEmpty(hearingType.getJudge()) && hearingType.getJudge().equals(existingJudgeCode)) {
+                hearingType.setJudge(requiredJudgeCode);
+                log.info(String.format("Judge's code in Case with ethosReference %s is updated", caseData.getEthosCaseReference()));
+            }
+        }
+    }
+
+//    public RefDataFixesData insertClaimServedDate(RefDataFixesDetails refDataFixesDetails, String authToken) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java:93]: This commented out code has nothing to do with the fix you're doing

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java (hunk 4 above)
Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/refdatafixes/ReferenceDataFixesController.java]: removed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java (hunk 5 above)
Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java:57]: removed throwing error

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java (hunk 6 above)
Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java]: returning list.of only now

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java (hunk 7 above)
Comment (gauravhmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java]: removed

**Review by jthmcts — APPROVED** [review]

**Review by btensay — APPROVED** [review]

**Pull request merged**